### DISCUSSION TITLE
Status code: 400

### DISCUSSION THREAD

**Discussion by psalmuel19** [cjbarrie/academictwitteR#209] [Q&A]
Hi all,

Having status code: 400 error running the code below. My suspicion is with the query list. Any help is appreciated.
```
tweets <-
  get_all_tweets(
    query = "presidential election, online disinformation, institutional mistrust, disinformation campaign OR propaganda, fake news, fact check, rumours, corruption OR looting, terrorism OR boko haram AND Fulani herdsmen, INEC OR electoral commission, credible election, INEC server, smart card reader OR SCR, permanent voters card OR PVC, electronic voters register OR EVR, third force, propaganda, fake account, social media, flawed democracy, card reader failure, manual accreditation, rigging AND underage voting, polling unit, money bag, trader moni, vote buying, collation centre AND returning officer AND presiding officer AND ballot box snatching, coalition, national security, security agents, NigeriaDecides, missing server result, presidential candidates, president Buhari, Buhari Media Centre OR BMC OR Buhari Media Organization, Atiku Abubakar, opposition party, Peoples Democratic Party OR PDP, All Progressives Congress OR APC OR ruling party, Muhammadu Buhari, #NigeriaDecides2019, NigeriaDecides, #LetsGetNigeriaWorkingAgain, #NeverAgain, #Atikulated, #BuhariMustGo, #NextLevel",
    start_tweets = "2018-01-10T00:00:00Z",
    end_tweets = "2019-31-03T00:00:00Z",
    bearer_token,
    data_path = "/Users/samuelola/Downloads/R/",
    bind_tweets = FALSE
  )
```

#45 #180 @justinchuntingho @cjbarrie

**Comment by chainsawriot** [comment]
Please read [this](https://developer.twitter.com/en/docs/twitter-api/v1/rules-and-filtering/search-operators) and help yourself: whether a comma is one of the operators there?

**Reply by psalmuel19** [reply]
Thank you for the response. I have checked the link and tried a couple of options. 
i. Removed the comma
ii. Added "" to each word with comma to separate the strings.
None worked.

**Reply by chainsawriot** [reply]
Please also pay attention to [the format of the date](https://developer.twitter.com/en/docs/twitter-api/tweets/search/api-reference/get-tweets-search-all), ISO 8601. `2019-31-03T00:00:00Z` is doomed to fail.

**Reply by psalmuel19** [reply]
Thank you. Corrected that.  So this is what I currently have:
```
tweets <-
  get_all_tweets(
    "online disinformation presidential election institutional mistrust disinformation campaign OR propaganda fact check corruption OR looting terrorism OR boko haram AND Fulani herdsmen INEC OR electoral commission credible election INEC server smart card reader OR SCR, permanent voters card OR PVC electronic voters register OR EVR third force propaganda fake social media account card reader failure manual accreditation rigging AND underage voting polling unit money bag trader moni vote buying collation centre AND returning officer AND presiding officer AND ballot box snatching national security security agents NigeriaDecides missing server result presidential candidates president Buhari Buhari Media Centre OR BMC OR Buhari Media Organization Atiku Abubakar opposition party Peoples Democratic Party OR PDP All Progressives Congress OR APC OR ruling party Muhammadu Buhari #NigeriaDecides2019 #LetsGetNigeriaWorkingAgain #NeverAgain #Atikulated BuhariMustGo #NextLevel",
    "2018-10-01T00:00:00Z",
    "2019-03-31T00:00:00Z",
    bearer_token,
    data_path = "/Users/samuelola/Downloads/R/",
    bind_tweets = FALSE
  )
```
This is the error I got:
**Error in make_query(url = endpoint_url, params = params, bearer_token = bearer_token,  : 
  something went wrong. Status code: 400**

**Reply by chainsawriot** [reply]
"online disinformation presidential election institutional mistrust disinformation campaign OR propaganda fact check corruption OR looting terrorism OR boko haram AND Fulani herdsmen INEC OR electoral commission credible election INEC server smart card reader OR SCR, permanent voters card OR PVC electronic voters register OR EVR third force propaganda fake social media account card reader failure manual accreditation rigging AND underage voting polling unit money bag trader moni vote buying collation centre AND returning officer AND presiding officer AND ballot box snatching national security security agents NigeriaDecides missing server result presidential candidates president Buhari Buhari Media Centre OR BMC OR Buhari Media Organization Atiku Abubakar opposition party Peoples Democratic Party OR PDP All Progressives Congress OR APC OR ruling party Muhammadu Buhari #NigeriaDecides2019 #LetsGetNigeriaWorkingAgain #NeverAgain #Atikulated BuhariMustGo #NextLevel"

1. `AND` is not an operator. However, " " is. (i.e. intersect). Therefore, you are making a lot of intersections in your query and it is almost impossible to find any meaningful results. Read the query guide.
2. How about a shorter query first?

**Reply by psalmuel19** [reply]
So I tried this: 
`tweets <-
  get_all_tweets(
    "online disinformation presidential election institutional mistrust disinformation campaign propaganda fact check terrorism INEC Independent electoral commission INEC server smart card reader permanent voters card electronic voters register third force OR coalition fake social media account card reader failure rigging trader moni vote buying NigeriaDecides Buhari Media Centre Peoples Democratic Party PDP All Progressives Congress APC ruling party Muhammadu Buhari #NigeriaDecides2019 #LetsGetNigeriaWorkingAgain #NeverAgain #Atikulated BuhariMustGo #NextLevel",
    "2018-10-01T00:00:00Z",
    "2019-03-31T00:00:00Z",
    bearer_token,
    data_path = "/Users/samuelola/Downloads/R/",
    bind_tweets = FALSE
  )`

No error. Thing is that it only returned ONE tweet.

**Reply by chainsawriot** [reply]
expected. Instead of I answer your question, how about I ask you 3 questions and you give me the answers?

1. What is the function of spaces in a query (e.g. as in the query `watching now`)?
2. How would Twitter interpret "online disinformation presidential election institutional mistrust disinformation campaign propaganda fact check terrorism INEC Independent electoral commission" as a search query? For example, whether it captures tweets with either "online", "information" or "disinformation"; or, it captures tweets containing "online", "disinformation" and "presidential" simultaneously, i.e. the intersect of all terms?
2. Do you think a 280-character tweet would contain 20 or more of your keywords simultaneously?

**Reply by psalmuel19** [reply]
Let me try.
1. The space is to ask it to find tweets containing both "watching" and "now"?
2. I think this goes into question 3. Again there appears to a problem with the way I have built the query. I see the point in the fact that a 280-character tweet would most likely not contain the keywords simultaneously.

Let me have another go at the query list once more. I appreciate your responses a lot.

**Reply by psalmuel19** [reply]
Hi, so I changed the query to just hashtags. Here is the code.

`tweets <-
  get_all_tweets(
    "(#NigeriaDecides2019) OR (#NigeriaDecides) OR (#LetsGetNigeriaWorkingAgain) OR (#NeverAgain) OR (#Atikulated) OR (BuhariMustGo) OR (#NextLevel)",
    "2018-10-01T00:00:00Z",
    "2019-03-31T00:00:00Z",
    bearer_token,
    data_path = "/Users/samuelola/Downloads/R/",
    bind_tweets = FALSE
  )`

Returned 492 tweets.

**Comment by psalmuel19** [comment]
Okay, let me try that now and revert. I feel I am getting closer

**Reply by psalmuel19** [reply]
It returned just ONE tweet.
